File: Back-end/src/main/java/KJX/KJX/controller/ScheduleController.java
package KJX.KJX.controller;

import KJX.KJX.entity.Schedule;
import KJX.KJX.service.ScheduleService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import java.util.List;

@RestController
@RequestMapping("/api/schedules")
@CrossOrigin(origins = "*")
public class ScheduleController {
    
    @Autowired
    private ScheduleService scheduleService;
    
    @GetMapping
    public ResponseEntity<List<Schedule>> getAllSchedules() {
        return ResponseEntity.ok(scheduleService.getAllSchedules());
    }
    
    @GetMapping("/{id}")
    public ResponseEntity<?> getScheduleById(@PathVariable Long id) {
        return scheduleService.getScheduleById(id)
            .map(ResponseEntity::ok)
            .orElseGet(() -> ResponseEntity.notFound().build());
    }
    
    @GetMapping("/day/{day}")
    public ResponseEntity<?> getScheduleByDay(@PathVariable String day) {
        return scheduleService.getScheduleByDay(day)
            .map(ResponseEntity::ok)
            .orElseGet(() -> ResponseEntity.notFound().build());
    }
    
    @GetMapping("/team/{teamId}")
    public ResponseEntity<List<Schedule>> getSchedulesByTeamId(@PathVariable Long teamId) {
        return ResponseEntity.ok(scheduleService.getSchedulesByTeamId(teamId));
    }
    
    @PostMapping
    public ResponseEntity<Schedule> createSchedule(@RequestBody Schedule schedule) {
        return ResponseEntity.ok(scheduleService.saveSchedule(schedule));
    }
    
    @PostMapping("/day")
    public ResponseEntity<Schedule> saveOrUpdateByDay(@RequestBody Schedule schedule) {
        return ResponseEntity.ok(scheduleService.saveOrUpdateByDay(schedule));
    }
    
    @PutMapping("/{id}")
    public ResponseEntity<Schedule> updateSchedule(@PathVariable Long id, @RequestBody Schedule scheduleDetails) {
        return ResponseEntity.ok(scheduleService.updateSchedule(id, scheduleDetails));
    }
    
    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteSchedule(@PathVariable Long id) {
        scheduleService.deleteSchedule(id);
        return ResponseEntity.noContent().build();
    }
}

File: nodejs-backend/src/controllers/schedule.controller.js
/**
 * Schedule Controller
 * Handles schedule management endpoints
 * Converted from Spring Boot ScheduleController
 */

const scheduleService = require('../services/schedule.service');

class ScheduleController {
  /**
   * GET /api/schedules
   * Get all schedules
   */
  async getAllSchedules(req, res, next) {
    try {
      const schedules = await scheduleService.getAllSchedules();
      res.json(schedules);
    } catch (error) {
      next(error);
    }
  }

  /**
   * GET /api/schedules/:id
   * Get schedule by ID
   */
  async getScheduleById(req, res, next) {
    try {
      const schedule = await scheduleService.getScheduleById(req.params.id);
      if (!schedule) {
        return res.status(404).json({
          success: false,
          message: 'Schedule not found'
        });
      }
      res.json(schedule);
    } catch (error) {
      next(error);
    }
  }

  /**
   * GET /api/schedules/day/:day
   * Get schedule by day
   */
  async getScheduleByDay(req, res, next) {
    try {
      const schedule = await scheduleService.getScheduleByDay(req.params.day);
      if (!schedule) {
        return res.status(404).json({
          success: false,
          message: 'Schedule not found'
        });
      }
      res.json(schedule);
    } catch (error) {
      next(error);
    }
  }

  /**
   * GET /api/schedules/team/:teamId
   * Get schedules by team ID
   */
  async getSchedulesByTeamId(req, res, next) {
    try {
      const schedules = await scheduleService.getSchedulesByTeamId(req.params.teamId);
      res.json(schedules);
    } catch (error) {
      next(error);
    }
  }

  /**
   * POST /api/schedules
   * Create new schedule
   */
  async createSchedule(req, res, next) {
    try {
      const schedule = await scheduleService.createSchedule(req.body);
      res.json(schedule);
    } catch (error) {
      next(error);
    }
  }

  /**
   * POST /api/schedules/day
   * Save or update schedule by day
   */
  async saveOrUpdateByDay(req, res, next) {
    try {
      const schedule = await scheduleService.saveOrUpdateByDay(req.body);
      res.json(schedule);
    } catch (error) {
      next(error);
    }
  }

  /**
   * PUT /api/schedules/:id
   * Update schedule
   */
  async updateSchedule(req, res, next) {
    try {
      const schedule = await scheduleService.updateSchedule(req.params.id, req.body);
      res.json(schedule);
    } catch (error) {
      if (error.message.includes('not found')) {
        return res.status(404).json({
          success: false,
          message: error.message
        });
      }
      next(error);
    }
  }

  /**
   * DELETE /api/schedules/:id
   * Delete schedule
   */
  async deleteSchedule(req, res, next) {
    try {
      await scheduleService.deleteSchedule(req.params.id);
      res.status(204).send();
    } catch (error) {
      if (error.message.includes('not found')) {
        return res.status(404).json({
          success: false,
          message: error.message
        });
      }
      next(error);
    }
  }
}

module.exports = new ScheduleController();
